### PR TITLE
Add more time details about writing the raft engine

### PR DESCRIPTION
**Pull request by sticnarf**
Addition to https://github.com/pingcap/kvproto/pull/926.

The more detailed time breakdown can be recorded and returned to the client.

Some field indexes can be changed because they are never used in TiKV yet.

### CODE REVIEW

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -177,10 +175,8 @@ message PessimisticLockResponse {
     // In legacy TiKV, this field is not used even 'force' is false. In that case, an empty value indicates
     // two possible situations: (1) the key does not exist. (2) the key exists but the value is empty.
     repeated bool not_founds = 6;
-    // Scan details during constraint checking.
-    ScanDetailV2 scan_detail_v2 = 7;
-    // Raftstore writing durations of the request. Not exists if in-memory pessimistic locking takes effect.
-    WriteDetail write_detail = 8;
+    // Execution details about the request processing.
+    ExecDetailsV2 scan_detail_v2 = 7;
```
Comment (you06) [inline proto/kvrpcpb.proto:179]: Name it with `exec_details_v2`?

File: proto/kvrpcpb.proto [hunk 2]
```diff
@@ -298,10 +294,8 @@ message CommitResponse {
     KeyError error = 2;
     // If the commit ts is derived from min_commit_ts, this field should be set.
     uint64 commit_version = 3;
-    // Scan details during constraint checking.
-    ScanDetailV2 scan_detail_v2 = 4;
-    // Raftstore writing durations of the request. Not exists if in-memory pessimistic locking takes effect.
-    WriteDetail write_detail = 5;
+    // Execution details about the request processing.
+    ExecDetailsV2 scan_detail_v2 = 4;
```
Comment (you06) [inline proto/kvrpcpb.proto:298]: ditto

File: proto/kvrpcpb.proto (hunk 1 above)
Comment (sticnarf) [reply proto/kvrpcpb.proto]: Oops.. Fixed now.

**Review by you06 — APPROVED** [review]
LGTM, what about `ResolveLockResponse`, no details in it yet.

**Comment by sticnarf** [comment]
> LGTM, what about `ResolveLockResponse`, no details in it yet.

I don't add these to non-critical requests yet. I can add them all

**Comment by you06** [comment]
@sticnarf is this PR ready to merge?

**Comment by sticnarf** [comment]
> @sticnarf is this PR ready to merge?

Let's merge it.

**Pull request merged**